[PANC-71] Create mock Airtel Api.

sanchit-0 opened the pull request:
## Description

[PANC-71 Write Mock APIs for Airtel money](https://fynarfin.atlassian.net/browse/PANC-71).

This PR creates Mock Apis for following Airtel Apis:
- [Payments - USSD Push](https://developers.airtel.africa/documentation/collection-apis/2.0)
- [Transaction Enquiry](https://developers.airtel.africa/documentation/collection-apis/2.0)


The API specs can be found [here](https://docs.google.com/document/d/1DbgIMHDKld5if_t6QdHWK7XjZmcXKNedDCtH4EpbeUI/edit?usp=sharing).


## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!
- [x] Followed the PR title naming convention mentioned above.

- [x] Design related bullet points or design document link related to this PR added in the description above.

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [x] Created/updated unit or integration tests for verifying the changes made.

- [ ] Added required Swagger annotation and update API documentation with details of any API changes if applicable

- [x] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

sanchit-0 commented on src/main/java/org/mifos/connector/airtel/mockairtel/service/AirtelMockService.java:
```diff
@@ -0,0 +1,154 @@
+        if (transactionId.equals(transactionIdSuccessful)) {
+            message = TransferStatus.SUCCESSFUL.name();
+            status = TransactionStatus.TS.name();
+            code = HttpStatus.OK.toString();
```
The code now has the value 200 Ok instead of 200 is that fine? Otherwise I think we have to create a seperate enum for response codes.

sanchit-0 commented on build.gradle:
```diff
@@ -41,7 +41,7 @@ dependencies {
     testImplementation 'org.springframework.boot:spring-boot-starter-test'
     testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
-    implementation 'org.mifos:ph-ee-connector-common:1.9.1-SNAPSHOT'
+    implementation 'org.mifos:ph-ee-connector-common:1.10.0-SNAPSHOT'
```
We are using `1.10.0` as connector common [PR](https://github.com/fynarfin/ph-ee-connector-common/pull/1) got merged and all the Airtel related DTOs are in `1.10.0`.

logoutdhaval commented on src/main/java/org/mifos/connector/airtel/mockairtel/utils/TransactionStatus.java:
```diff
@@ -0,0 +1,5 @@
+package org.mifos.connector.airtel.mockairtel.utils;
+
+public enum TransactionStatus {
```
Why this enum is required? You can use TransferStatus enum here.

sanchit-0 replied:
Done.

logoutdhaval commented on src/main/java/org/mifos/connector/airtel/mockairtel/api/definition/AirtelMockApi.java:
```diff
@@ -0,0 +1,36 @@
+        return airtelMockController.getTransactionStatus(transactionId);
+    }
+
+    @PostMapping("/merchant/v2/payments/")
```
remove this extra '/' from the endpoint

sanchit-0 replied:
Done.

ankita10r reviewed (commented):
LGTM

sanchit-0 commented on build.gradle:
```diff
@@ -31,7 +31,11 @@ ext {
     mavenCentral()
     maven {
-        url = uri('https://jfrog.sandbox.fynarfin.io/artifactory/fyn-libs-snapshot')
+        url = uri('https://fynarfinhub.jfrog.io/artifactory/ph-ee-repo-local')
+        credentials {
+            username = "maven-repo-user"
+            password = "123QWEasd"
+        }
```
These changes have been done as jfrog is not working rn.

logoutdhaval replied:
okay, remove this part once jfrog issue is fixed.

sanchit-0 replied:
Okay

logoutdhaval commented on build.gradle:
```diff
@@ -41,7 +45,7 @@ dependencies {
     testImplementation 'org.springframework.boot:spring-boot-starter-test'
     testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
-    implementation 'org.mifos:ph-ee-connector-common:1.9.1-SNAPSHOT'
+    implementation 'org.mifos:ph-ee-connector-common:0.0.0'
```
Also, update this to the released version after the jfrog issue is fixed.

sanchit-0 replied:
Okay

sanchit-0 replied:
Using 0.0.0 as this [PR](https://github.com/fynarfin/ph-ee-connector-common/pull/3) has not been merged and new version is not released.

sanchit-0 committed:
Create mock Airtel APIs.
The response codes in the `application.yml` file have been added from the Airtel documentation.

Similarly, the responses returned by the API are based on the Airtel documentation.
